mipengine/node/udfgen/udfgenerator.py

```diff
@@ -633,7 +639,8 @@ def generate_udf_select_stmt(
 
     from_subexpr = prettify(SEP.join(table_names))
 
-    nodeid_column = " $node_id" + AS + "node_id"
+    node_id_type = "varchar(50)"
+    nodeid_column = f" CAST('$node_id' AS {node_id_type})" + AS + "node_id"
```

Comment:
Shouldn't this be as well
node_id_type = __convert_to_monetdb_column_type('text')
?